lib/assert.js

```diff
@@ -278,7 +278,13 @@ function expectedException(actual, expected) {
     // Ignore.  The instanceof check doesn't work for arrow functions.
   }
 
-  return expected.call({}, actual) === true;
+  try {
+    return expected.call({}, actual) === true;
+  } catch (e) {
+    // Ignore. If expected is a class constructor, an error will be thrown
+  }
```

Comment:
For what it's worth, and likely not telling anyone anything they don't already know, the "proper" fix is probably something like:

* Unlock API
* Change function signature to add an options object and the user can send a validation option and/or a constructor option for an object that the thrown error must match
* Relock the API

That could be done in a semver-minor way if we leave the legacy behavior intact, bug and all.

I'm not necessarily recommending that route--the API was locked for a reason--but just saying: If we want to fix it "right", the solution probably looks something like that...